YubiKit/YubiKit/Helpers/Additions/YKFNSStringAdditions.m

```diff
@@ -17,30 +17,48 @@
 @implementation NSString(NSString_OATH)
 
 - (void)ykf_OATHKeyExtractPeriod:(NSUInteger *)period issuer:(NSString **)issuer account:(NSString **)account label:(NSString **)label {
-    NSString *key = self;
+    NSString *periodIssuer;
+    NSMutableArray *componentsArray;
     
+    BOOL noIssuer = FALSE;
+
+    if ([self containsString: @":"]) {
+        // TOTP key with format [period]/[issuer]:[account]
+        NSArray *labelComponents = [self componentsSeparatedByString:@":"];
```

Comment:
nit: there is optionally another way - to check only first occurrence and then substrings from this index.
Example:
NSRange range = [time rangeOfString:@":"];
NSString *issuer = [time substringToIndex:range.location];
and using substringFromIndex to get account.

And IMHO it's more logical to start parsing period first to get separated label + period, rather than periodIssuer + account.